components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/step/impl/DefaultStepHandler.java

```diff
@@ -897,6 +897,14 @@ protected String getRedirectUrl(HttpServletRequest request, HttpServletResponse
                 }
             }
         }
+        if (StringUtils.isBlank(reCaptchaParamString.toString())) {
+            String captchaParamStringFromContext = (String) context.getProperty(
+                    FrameworkConstants.CAPTCHA_PARAM_STRING);
+            if (captchaParamStringFromContext != null) {
+                reCaptchaParamString.append(captchaParamStringFromContext);
```

Comment:
Are we certain that the `captchaParamStringFromContext` is not going to be a blank string? If not, let's go with !isBlank() check to be safe.

Reply:
use StringUtils to check for null and blank